libs/design-system/navigation/header/src/lib/header.component.ts

```diff
@@ -125,6 +130,15 @@ export class HeaderComponent {
     });
 
     explicitEffect([this.menuOffsetPx], () => this.updateMenuPositions(), { defer: true });
+
+    this.router.events
```

Comment:
Inject router here since it is the only place it is used. Also it should be optional so the header works in apps without the router.
```ts
inject(Router, { optional: true })?.events...
```